packages/components/src/components/form/bal-select/bal-select.tsx

```diff
@@ -384,11 +396,16 @@ export class Select {
    ********************************************************/
 
   private get optionArray() {
-    return Array.from(this.options, ([_, value]) => value).filter(option => {
-      if (this.typeahead) {
+    const options = Array.from(this.options, ([_, value]) => value)
+    if (!this.typeahead || this.remote) {
+      return options
+    }
+
+    return options.filter(option => {
+      if (this.filter === 'includes') {
```

Comment:
could be changed to:
  return (this.filter === 'includes') ? includes(...) : startsWith(...)

Reply:
done thanks